tests/integration/deps/test_podman_compose_deps.py

```diff
@@ -183,3 +185,74 @@ def test_deps_fails(self):
                 compose_yaml_path(suffix),
                 "down",
             ])
+
+
+class TestComposeConditionalDepsHealthy(unittest.TestCase, RunSubprocessMixin):
+    def test_up_deps_healthy(self):
+        suffix = "-conditional-healthy"
+        try:
+            self.run_subprocess_assert_returncode([
+                podman_compose_path(),
+                "-f",
+                compose_yaml_path(suffix),
+                "up",
+                "sleep",
+                "--detach",
+            ])
+
+            # The `podman wait --condition=healthy` is invalid prior to 4.6.0.
+            # Since the podman-compose project uses podman 4.3.1 in github actions, we
+            # use sleep as workaround to wait until the `sleep` container becomes running.
+            time.sleep(3)
+
+            # self.run_subprocess_assert_returncode([
+            #     "podman",
+            #     "wait",
+            #     "--condition=running",
+            #     "deps_web_1",
+            #     "deps_sleep_1",
+            # ])
+
+            # check both web and sleep are running
+            output, _ = self.run_subprocess_assert_returncode([
+                podman_compose_path(),
+                "-f",
+                compose_yaml_path(),
+                "ps",
+                "--format",
+                "{{.ID}}\t{{.Names}}\t{{.Status}}\t{{.StartedAt}}",
+            ])
+            # extract container id of web
+            decoded_out = output.decode('utf-8')
+            web_cnt_id = ""
+            web_cnt_name = ""
+            web_cnt_status = ""
+            web_cnt_started = ""
+            sleep_cnt_id = ""
+            sleep_cnt_name = ""
+            sleep_cnt_started = ""
+            for line in decoded_out.split("\n"):
+                if "web" in line:
+                    web_cnt_id, web_cnt_name, web_cnt_status, web_cnt_started = line.split("\t")
+                if "sleep" in line:
+                    sleep_cnt_id, sleep_cnt_name, _, sleep_cnt_started = line.split("\t")
+            self.assertNotEqual("", web_cnt_id)
```

Comment:
If asserts lived in the loop then the code would be way simplier.

Reply:
Moving asserts into the loop isn't quite right. I simplify the code in other way by removing the loop and use list comprehension.